MusicHub/tracks/serializers.py

```diff
@@ -0,0 +1,50 @@
+from rest_framework import serializers
+from rest_framework.serializers import (
+    ModelSerializer,
+    ValidationError,
+)
+from tinytag import TinyTag
+
+from MusicHub.main.utils import format_to_minutes
+from MusicHub.tracks.models import Track
+
+
+class CreateTrackSerializer(ModelSerializer):
+    class Meta:
+        model = Track
+        fields = ["filename", "track", "public"]
+
+    def validate_track(self, value):
+        if value.size > 30000000:
+            raise ValidationError("File cannot be bigger than 30 Mb")
+        return value
+
+    def create(self, validated_data):
+
+        track = Track.objects.create(
+            created_by=self.context.get("user"), **validated_data
+        )
+        track.track_length = TinyTag.get(track.track.path).duration
```

Comment:
I belive naming `file` instead of `track` in `Track` model better. For instance here `track.file.path` instead of `track.track.path`